### DISCUSSION TITLE
Unexpected behavior when setting IsShareSizeScope and ShareSizeGroup

### DISCUSSION THREAD

**Discussion by Flithor** [AvaloniaUI/Avalonia#10927] [Q&A] [answered]
**Describe the bug**  
I'm writing a HeaderedContentControl based control, it is designed for automatic layout of a form, and adaptively handles the display width of field area for each column under the same Panel.  
Previously it was a control I wrote under WPF ([xaml](https://github.com/Flithor/ReusableCodes/blob/main/WPF/FormField.xaml),[code](https://github.com/Flithor/ReusableCodes/blob/main/WPF/FormField.cs)), and now after porting to Avalonia, I found that the behavior is inconsistent with that in WPF.  
It seems to be caused by the logical order when calculating the layout. I can't figure out where the problem is. I hope to get help.

**To Reproduce**
The avalonia version control code here: ([axaml](https://github.com/Flithor/ReusableCodes/blob/main/AvaloniaUI/FormField.axaml),[code](https://github.com/Flithor/ReusableCodes/blob/main/AvaloniaUI/FormField.axaml.cs))  
Check the Preview in axaml.
**NOTICE:** The preview of the control in `Design.PreviewWith` is not exactly the same as that in the window, please copy it to the window to test the actual layout.
The behavior in`Grid` and `StackPanel` is okay, but there is something wrong in `ItemsControl`, set the ` Grid.IsSharedSizeScope="True"` on `ItemsControl` will works (it's existed in code, try remove it and check the behavior.)

**Expected behavior**  
The layout should be update automatically after the panel is loaded, and it doesn't need to set `IsSharedSizeScope` outside externally.

**Screenshots**
![image](https://user-images.githubusercontent.com/23412916/230251243-5399611a-36f4-4d40-a0cb-bf65d4c5b34a.png)

**Desktop (please complete the following information):**

- OS: Windows 11
- Version 0.10.18

**Additional context**
Some control behavior in `Design.PreviewWith` display in designer is not same in window, may need to check.

**Comment by timunie** [comment, marked as answer]
In Avalonia we have visual and logical parent. These are similar, but not the same. You can use [DevTools](https://docs.avaloniaui.net/docs/getting-started/developer-tools) to check both trees. I think you are looking for the logical parent. In case of `ItemsControl` I think the type of parent you are looking for is `ItemsControl` and not `Panel`. 

It works for me if I change your code like this:

```cs
private AvaloniaObject? GetPanelParent(StyledElement control)
{
    return this.GetLogicalAncestors().FirstOrDefault(x => x is Panel or ItemsControl) as AvaloniaObject;
    
    // Note this works also, but I shortened for readability. 

    // StyledElement? parent = control;
    // do
    // {
    //     if (parent is Panel or ItemsControl) 
    //         return parent;
    //     
    //     parent = parent.Parent;
    // } while (parent != null);
    //
    // return null;
}
```

**Result:** 

![image](https://user-images.githubusercontent.com/47110241/230313382-eea11403-44ae-418f-a252-f5261a282892.png)

I don't think it is an actual bug, so I'll convert this into a Q&A. I hope you're fine with this.

**Reply by Flithor** [reply]
Looks only the control in "logical tree" will make the `IsSharedSizeScope` works.
I don't know that, so just try in same way like WPF to find the panel on "visual tree".
